### PR TITLE
ci(pie-monorepo): DSW-1487 Add changesets snapshot functionality.

### PR DESCRIPTION
**Pull request by siggerzz**
This PR adds Changeset snapshot release functionality, and is a slight modification to the workflow seen in [Shopify's Polaris repo](https://github.com/Shopify/polaris/blob/main/.github/workflows/snapit.yml).

### CODE REVIEW

File: .github/workflows/changeset-snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+      - name: Create an .npmrc
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: |
+          cat << EOF > "$HOME/.npmrc"
+            //registry.npmjs.org/:_authToken=$NPM_TOKEN
+          EOF
+
+      - name: Get Markdown Changeset Files Difference
```
Comment (siggerzz) [inline .github/workflows/changeset-snapshot.yml:77]: `Get Markdown Changeset Files Difference` and `Delete Unrelated Markdown Files in Changeset` are the only additions I've made to this file (apart from refactoring to work with our repo).


Unlike Shopify's implementation, this will prevent a snapshot being created for any changesets that are currently in the `main` branch.

**Review by raoufswe — APPROVED** [review]

**Review by xander-marjoram — APPROVED** [review]
